### PR TITLE
Blog - Process data in models

### PR DESCRIPTION
**Pull request by Emmy-github-webdev**
## Blog - Process data in models

In this project I
- Create users
- Create posts
- Create comments
- Add a method that returns the 3 most recent posts for a given user.
- Add a  method that updates the posts counter for a user.
- Add a method that returns the 5 most recent comments for a given post.
- Add a method that updates the comments counter for a post.
- Add a method that updates the likes counter for a post.
- Create a method called current_user in your ApplicationController.

### CODE REVIEW

File: app/models/blog.rb [hunk 1]
```diff
@@ -1,5 +1,13 @@
 class Blog < ApplicationRecord
-  has_many :comment
   belongs_to :user
-  has_many :like
+  has_many :comments
+  has_many :likes
+
```
Comment (MahmoudBakr23) [inline app/models/blog.rb]: - [ ] **OPTIONAL**
Please consider using the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` in order to call the `update_post` method every time a new post has been saved, then make the method private

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :blog
+
```
Comment (MahmoudBakr23) [inline app/models/comment.rb:5]: - [ ] **OPTIONAL**
Please consider using the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` in order to call the `update_comment` method every time a new comment has been saved, then make the method private

File: app/models/like.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 class Like < ApplicationRecord
-  belongs_to :blog
   belongs_to :user
+  belongs_to :blog
+
```
Comment (MahmoudBakr23) [inline app/models/like.rb:5]: - [ ] **OPTIONAL**
Please consider using the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` in order to call the `update_likes` method every time a new like has been saved, then make the method private

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Emmy-github-webdev 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

**OPTIONAL**

- The files changed are very big, please add any dir that's should not be committed to the `.gitignore` file.
- Kindly fix the broken links in your README file for a professional repo.
- _Consider the comments down below._

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 2 above)
Comment (Emmy-github-webdev) [reply app/models/comment.rb:5]: Recommendation implemented. Thank you

**Pull request merged**